portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -0,0 +1,102 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+////////////////////////////////
+////  new packages imported
+////////////////////////////////
+
+import com.google.gson.Gson;
+
+@WebServlet("/login")
+public class LoginServlet extends HttpServlet {
+
+  private class LoginStatus{
+
+      private boolean isLoggedIn;
+      private String logInURL;
+      private String logOutURL;
+      private String userEmail;
+
+      public LoginStatus(boolean nIsLoggedIn, String logInURL,String logOutURL,String userEmail){
+          this.isLoggedIn = nIsLoggedIn;
+          this.logInURL = logInURL;
+          this.logOutURL = logOutURL;
+          this.userEmail = userEmail;
+      }
+  }
+
+  private static final Gson aGson = new Gson();
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    response.setContentType("application/json;");
+    UserService userService = UserServiceFactory.getUserService();
+    
+    String json;
+
+    if (userService.isUserLoggedIn()) 
+    {
+        LoginStatus aLoginStatus = new LoginStatus(
+            true,
+            "",
```

Comment:
To make reviewer's life easier, consider comment the names for arguments that is not obvious. For example: 
```suggestion
            /* nIsLoggedIn= */ true,
            /* logInURL= */ "",
```